src/jdk.jdi/share/classes/com/sun/tools/jdi/EventRequestManagerImpl.java

```diff
@@ -880,9 +877,8 @@ public void deleteEventRequest(EventRequest eventRequest) {
     public void deleteEventRequests(List<? extends EventRequest> eventRequests) {
         validateMirrors(eventRequests);
         // copy the eventRequests to avoid ConcurrentModificationException
-        Iterator<? extends EventRequest> iter = (new ArrayList<>(eventRequests)).iterator();
-        while (iter.hasNext()) {
-            ((EventRequestImpl)iter.next()).delete();
+        for (EventRequest eventRequest : new ArrayList<>(eventRequests)) {
+            ((EventRequestImpl) eventRequest).delete();
```

Comment:
I'm not sure the `eventRequest` needs to be wrapped into new `ArrayList`. I think, this should work without it.

Reply:
It seems odd at first, but seems to have been done purposefully. Perhaps it has to do with the fact that delete() is being called, which does appear to also end up deleting the request from the list (see line 167). I don't think you want to be deleting from or adding to a collection that you are iterating over.

Reply:
You are right, thanks.